File: backend/src/main/java/com/example/backend/entity/Category.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "categories")
public class Category {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    private String name;
    private String description;
    private String imageUrl;
    private Boolean active = true;
}

File: front/lib/models/category.dart
class Category {
  final int id;
  final String name;
  final String description;
  final String? imageUrl;
  final bool active;

  Category({
    required this.id,
    required this.name,
    required this.description,
    this.imageUrl,
    required this.active,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      imageUrl: json['imageUrl'],
      active: json['active'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'imageUrl': imageUrl,
      'active': active,
    };
  }
}